packages/cloudflare/src/cli/build/bundle-server.ts

```diff
@@ -17,6 +17,20 @@ import { normalizePath, patchCodeWithValidations } from "./utils/index.js";
 /** The dist directory of the Cloudflare adapter package */
 const packageDistDir = path.join(path.dirname(fileURLToPath(import.meta.url)), "../..");
 
+/**
+ * List of optional Next.js dependencies.
+ * They are not required for Next.js to run but only needed to enabled specific features.
+ * When one of those dependency is required, it should be installed by the application.
+ */
+const optionalDependencies = [
+  "caniuse-lite",
+  "critters",
+  "jimp",
+  "probe-image-size",
+  // `server.edge` is not available in react-dom@18
+  "react-dom/server.edge",
```

Comment:
how come we're no longer going to change this require to use `react-dom/server.browser` for <= 18?

Reply:
I did read the description, but it wasn't clear to me why the re-exported version doesn't need this

Reply:
This is what I tried to explain in the PR description: `pages.runtime.prod.js` does that (first snippet), the actual build failure was only caused by the re-export of `react-dom/server.edge` (second snippet) which is patched by the optional deps (third snippet).

I have tested with the versions from the original issue and it works (fails without the patch).

Does this make sense?

Reply:
Are we saying that the re-exported version is never actually used for the app then?

Reply:
I think we don't need to know.

If some code is using the re-export then `require("react-dom/server.edge")` will throw in `pages.runtime.prod.js`.
We are respecting that behavior.

Either nobody is using it or that code have to use a try...catch

Reply:
Understood